### PR TITLE
feat(storage): use distinct count calculated by the xor hash function

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Xor filter calculates the hash value for each value in the calculation process. Using this hash value, we can get an approximate number of unique values, which can be used to replace the distinct count function `calc_column_distinct_of_values`. This optimization can improve the performance of data insertion.

Closes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
Is there some performance test after this PR?

**Comment by lichuang** [comment]
test may need add some case that `gen_columns_statistics` second param is not None.

**Comment by b41sh** [comment]
> Is there some performance test after this PR?

The time to insert 100m values was reduced by about 4s, and the time to build `BlockStatistics` was reduced to less than 1s.

image

File: src/query/storages/index/src/bloom.rs [hunk 1]
```diff
@@ -134,9 +140,11 @@ impl BlockFilter {
                 }
 
                 let filter = filter_builder.build()?;
+                if let Some(len) = filter.len() {
```
Comment (dantengsky) [inline src/query/storages/index/src/bloom.rs:143]: does `filter.len()` return the NDV or just the number of values?

Comment (b41sh) [reply src/query/storages/index/src/bloom.rs:143]: the len is the number of distinct values.

Comment (dantengsky) [reply src/query/storages/index/src/bloom.rs:143]: :ok_hand:

**Review by dantengsky — APPROVED** [review]
LGTM

**Review by lichuang — APPROVED** [review]

**Pull request merged**